products/bash/apic-subscribe.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+#******************************************************************************
+# Licensed Materials - Property of IBM
+# (c) Copyright IBM Corporation 2020. All Rights Reserved.
+#
+# Note to U.S. Government Users Restricted Rights:
+# Use, duplication or disclosure restricted by GSA ADP Schedule
+# Contract with IBM Corp.
+#******************************************************************************
+
+DEBUG=true
+NAMESPACE="cp4i"
+RELEASE="ademo"
+PROVIDER_ORG="main-demo"
+CATALOG="main-demo-catalog"
+CONSUMER_ORG=""
+PRODUCT=""
+APP=""
+ENDPOINT_SECRET_NAME=""
+BASE_PATH=""
+
+function usage() {
+  echo "Usage: $0 -n <namespace> -r <APIC CR name> -o <provider org> -c <catalog> -u <consumer org> -p <product> -a <app> -e <endpoint secret name> -b <base path>"
+}
+
+while getopts "a:b:c:e:n:o:p:r:u:" opt; do
+  case ${opt} in
+  a)
+    APP="$OPTARG"
+    ;;
+  b)
+    BASE_PATH="$OPTARG"
+    ;;
+  c)
+    CATALOG="$OPTARG"
+    ;;
+  e)
+    ENDPOINT_SECRET_NAME="$OPTARG"
+    ;;
+  n)
+    NAMESPACE="$OPTARG"
+    ;;
+  o)
+    PROVIDER_ORG="$OPTARG"
+    ;;
+  p)
+    PRODUCT="$OPTARG"
+    ;;
+  r)
+    RELEASE="$OPTARG"
+    ;;
+  u)
+    CONSUMER_ORG="$OPTARG"
+    ;;
+  \?)
+    usage
+    exit
+    ;;
+  esac
+done
+
+if [ -z "${PRODUCT}" ]; then echo "PRODUCT must be specified using -p" ; exit 1 ; fi
+
+# If CONSUMER_ORG not set then default based on PROVIDER_ORG
+if [ -z "${CONSUMER_ORG}" ]; then CONSUMER_ORG=${PROVIDER_ORG}-corp; fi
+
+# If APP not set then default based on PRODUCT
+if [ -z "${APP}" ]; then APP=${PRODUCT}-app; fi
+
+TICK="\xE2\x9C\x85"
+CROSS="\xE2\x9D\x8C"
+
+client_id="599b7aef-8841-4ee2-88a0-84d49c4d6ff2"
+client_secret="0ea28423-e73b-47d4-b40e-ddb45c48bb0c"
```

Comment:
is the client id and secret the same each time? I would have assumed that a id would change everytime it is created?

Reply:
This is the standard client_id/client_secret that is used to interact with the APIM REST API and it doesn't change. Note it is not the same as the client_id/client_secret that are different for every application that is used to call the DDD/EEI endpoints and is extracted on lines 200/201.

Reply:
got you 👍